src/lib/core/overlay/position/viewport-ruler.ts

```diff
@@ -8,12 +9,23 @@ import {Injectable} from '@angular/core';
  */
 @Injectable()
 export class ViewportRuler {
-  // TODO(jelbourn): cache the document's bounding rect and only update it when the window
-  // is resized (debounced).
 
+  /** Cached document client rectangle. */
+  private _documentRect: ClientRect;
+
+  constructor(scrollDispatcher: ScrollDispatcher) {
+
```

Comment:
nit: I can't stand extra whitespace, you can leave it if you want but I'll definitely delete it if I ever edit this file :p